File: scala/raindrops/src/main/scala/Raindrops.scala
object Raindrops {
  def convert(n: Int): String = {
    val result = Map(3 -> "Pling", 5 -> "Plang", 7 -> "Plong")
      .foldLeft("") { case (acc, (factor, s)) =>
        if (n % factor == 0) acc + s else acc
      }
    if (result.isEmpty) n.toString else result
  }
}

File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
class Raindrops {
    companion object {
        @JvmStatic
        fun convert(input: Int): String = mapOf(
                Pair(3, "Pling"),
                Pair(5, "Plang"),
                Pair(7, "Plong")
        ).map { pair ->
            if (input % pair.key == 0) pair.value else ""
        }.fold(input.toString()) { total, next ->
            total + next
        }.partition { ch ->
            ch.isLetter()
        }.toList().first { it.isNotEmpty() }
    }
}
